### DISCUSSION TITLE
Offscreen canvas

### DISCUSSION THREAD

**Discussion by ikiselev1989** [excaliburjs/Excalibur#2730] [Q&A]
is it possible to use offscreen canvas or something else from webworker stuff?

**Comment by eonarheim** [comment]
Hi @ikiselev1989, good question!

We don't have anything built in at the moment to support OffscreenCanvas and WebWorkers, but using them should be possible.

What type of problems are you looking to solve?

**Reply by ikiselev1989** [reply]
Hi @eonarheim 
Actually I have faced some performance issue. We are creating a project with a lot of graphics for the HiDPI screen. I got a screen freeze before an actor became visible. There are 3 layers with parallax.
So, I'm looking for a OffscreenCanvas for render.

**Reply by eonarheim** [reply]
@ikiselev1989 I'm curious where your performance bottle neck is, can you run a Chrome of Firefox performance trace? If you could post a trace that would help debug a lot.

<img width="325" alt="image" src="https://github.com/excaliburjs/Excalibur/assets/612071/3798f441-6d7f-4712-b618-7166b9d851cb">

<img width="297" alt="image" src="https://github.com/excaliburjs/Excalibur/assets/612071/94950142-254a-4387-9259-ddb3d260b76d">

Are you experiencing consistent performance degradation, or at specific times? Does removing the parallax feature improve perf?

**Reply by ikiselev1989** [reply]
@eonarheim 
[Trace-20230816T212502.zip](https://github.com/excaliburjs/Excalibur/files/12362446/Trace-20230816T212502.zip)

**Reply by ikiselev1989** [reply]
This happens when an animated sprite appears in the viewport. Removing parallax doesn't fix anything
